shard.go

```diff
@@ -180,14 +179,31 @@ func marshalValues(values map[uint8]interface{}) []byte {
 		}
 
 		// Encode value after field id.
-		// TODO: Support non-float types.
 		switch v := v.(type) {
 		case float64:
+			buf = make([]byte, 9)
 			binary.BigEndian.PutUint64(buf[1:9], math.Float64bits(v))
+		case bool:
+			buf = make([]byte, 2)
```

Comment:
It seems like I could preform the type conversion using the metastore around here:

https://github.com/influxdb/influxdb/blob/master/server.go#L1839

so the low-level conversion to float (and support for other types at this low-level as implied by the comments) may be unnecessary.